src/HotChocolate/Spatial/src/Types/GeoJSONInputObject.Parsing.cs

```diff
@@ -0,0 +1,165 @@
+using System.Collections.Generic;
+using HotChocolate.Language;
+using HotChocolate.Utilities;
+using NetTopologySuite.Geometries;
+
+namespace HotChocolate.Types.Spatial
+{
+    public abstract partial class GeoJSONInputObjectType<T>
+    {
+        protected void ValidateGeometryKind(ObjectValueNode valueNode, int fieldIndex)
+            => ParseGeometryKind(valueNode, fieldIndex);
+
+        protected GeoJSONGeometryType ParseGeometryKind(
+            ObjectValueNode valueNode,
+            int fieldIndex)
+        {
+            if (fieldIndex == -1)
+            {
+                throw ThrowHelper.InvalidStructure_TypeIsMissing(this);
+            }
+
+            IValueNode typeValue = valueNode.Fields[fieldIndex].Value;
+
+            if (!(_typeField.Type.ParseLiteral(typeValue) is GeoJSONGeometryType type))
```

Comment:
we now have is not

Reply:
i know, but we need the coordinates reference later